### ISSUE TITLE
use phpipam provider for multiple resources with count =

### ISSUE THREAD

**Issue by gricuk** [lord-kyron/terraform-provider-phpipam#3]
What about the possibility for adding multiple resources e.g
```
resource "phpipam_address" "ip" {
    count = 3
}
```

**Comment by lord-kyron** [comment]
Hi,
yes this is not implemented yet.
I've just forked the original developer work and made it work on the latest terraform.

I will take a look at this and check if I can add this feature to the next release!
Thank you!

**Comment by dogriverrat** [comment]
Do you have an example of using this provider with for_each ?    I tried using this:

```
#===============================================================================
# Get the first available address
#===============================================================================

data "phpipam_first_free_address" "next_address" {
  for_each    = var.vm_hostname_metadata
  subnet_id = data.phpipam_subnet.subnet.subnet_id
}

#===============================================================================
# phpIPAM Resource
# Reserve the address. Note that we use ignore_changes here to ensure that we
# don't end up re-allocating this address on future Terraform runs.
#===============================================================================

resource "phpipam_address" "ipaddress" {
  for_each    = var.vm_hostname_metadata
  subnet_id   = data.phpipam_subnet.subnet.subnet_id
  ip_address  = data.phpipam_first_free_address.next_address[each.key].ip_address
  hostname    = each.value.name
  description = "Managed by Terraform"

  lifecycle {
    ignore_changes = [
      "subnet_id",
      "ip_address",
    ]
  }
}
```
Unfortunately, the above fails after approving to make the change with the following:

```
<snip>
  Enter a value: yes

phpipam_address.ipaddress["UiSql"]: Creating...
phpipam_address.ipaddress["UiOrch"]: Creating...
phpipam_address.ipaddress["UiReport"]: Creating...
phpipam_address.ipaddress["UiReport"]: Creation complete after 0s [id=85]

Error: Error from API (409): IP address already exists

  on vsphere-windows-3x-with-ansible.tf line 114, in resource "phpipam_address" "ipaddress":
 114: resource "phpipam_address" "ipaddress" {



Error: Error from API (409): IP address already exists

  on vsphere-windows-3x-with-ansible.tf line 114, in resource "phpipam_address" "ipaddress":
 114: resource "phpipam_address" "ipaddress" {
```

So 1 IP is setup in phpIPAM but the other two resource instances fail as they are both attempting to use the same IP address.  For for context, I'm trying to use these IPs within my ```resource "vsphere_virtual_machine" "standalone"``` block which contains this:

```
resource "vsphere_virtual_machine" "standalone" {
  for_each         = var.vm_hostname_metadata
  name             = each.value.name

  <snip>

  clone {

    <snip>

     network_interface {
       ipv4_address = phpipam_address.ipaddress[each.key].ip_address
       ipv4_netmask = each.value.mask
     }

     <snip>
```

Since my plan hasn't hit that yet I don't think this it is relevant but more of context of what I'm hoping to do.  And I'm a Terraform newbie as well !!th

Thank you for any reply.  Happy New Year !!

**Comment by jordips** [comment]
@gricuk @lord-kyron  Just to inform,

PHPIPAM doesn't have an API method to reserve IPs right now. I tried some workaround in this issue:

https://github.com/paybyphone/terraform-provider-phpipam/issues/7#issuecomment-590394393

But is not a solution. Maybe you find a better workaround.

**Comment by lord-kyron** [comment]
We've released a new version - official 1.0 which major change is that it supports "count" now.
However, be sure to read the version changelog before updating!
